### PR TITLE
feat(settings): allow user to set settings for each query individually

### PR DESCRIPTION
**Pull request by janosh**
Adds ability to specify index settings programmatically. Addresses #16.

### CODE REVIEW

**Comment by Haroenv** [comment]
I just added a prettierrc, that way the formatting doesn't change too much, can you run prettier? thanks!

File: gatsby-node.js [hunk 1]
```diff
@@ -55,6 +55,10 @@ exports.onPostBuild = async function(
 
     await Promise.all(chunkJobs);
 
+    tmpIndex.setSettings(settings, err => {
```
[the remark this answers is not in the document]
Comment (janosh) [reply gatsby-node.js]: You mean like this? (Not sure what your second point is saying.)

```suggestion
if (settings) {
  tmpIndex.setSettings(settings);
}
```

[the remark this answers is not in the document]
Comment (Haroenv) [reply gatsby-node.js]: yep, that's what I meant!

File: gatsby-node.js [hunk 2]
```diff
@@ -55,6 +55,10 @@ exports.onPostBuild = async function(
 
     await Promise.all(chunkJobs);
 
+    if (settings) {
+      tmpIndex.setSettings(settings);
```
Comment (Haroenv) [inline gatsby-node.js]:
```suggestion
      indexToUse.setSettings(settings);
```

This way it should also work if there's no index yet. I'll add some tests to this

**Pull request merged**